notebooks/02 EDA.ipynb

```diff
@@ -0,0 +1,1519 @@
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Extract columns with 100% empty values\n",
+    "empty_cols = sorted_datetimes_df.isnull().sum() == sorted_datetimes_df.shape[0]\n",
+    "empty_cols = empty_cols[empty_cols].index.array"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1650383864082
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "for col in empty_cols:\n",
+    "    print(f\"{col} values: {sorted_datetimes_df[col].unique()}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1650383877149
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# df_remove_empty_cols\n",
+    "remove_empty_cols_df = sorted_datetimes_df.drop(columns=empty_cols)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1650383878041
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "remove_empty_cols_df[remove_empty_cols_df.stroke_ward_stay.notna()].shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1650383878750
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "remove_empty_cols_df.stroke_ward_stay.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "## Identify early block of missing data\n",
+    "\n",
+    "The first ~100k rows are missing a significant amount of data - is this due to the introduction of a new system?\n",
+    "\n",
+    "1. When does the block of missing data end? Does this align with a new clinical system?\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1650383897155
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# find the first valid index (ie. not null) for the presenting_complaint field, one of the frequently null fields\n",
```

Comment:
I'd add a comment here noting how the dataframe is sorted as the sort is done earlier.